src/Model_Hydro/Hydro_Init_ByFunction_AssignData.cpp

```diff
@@ -248,8 +248,9 @@ void Hydro_Init_ByFunction_AssignData( const int lv )
             Init_Function_User_Ptr( fluid_sub, x, y, z, Time[lv], lv, NULL );
 
 //          modify the initial condition if required
+//          --> always set the magnetic energy to zero since fluid_sub[ENGY] doesn't include that
             if ( OPT__RESET_FLUID )
-               Flu_ResetByUser_Func_Ptr( fluid_sub, x, y, z, Time[lv], 0.0, lv, NULL );
+               Flu_ResetByUser_Func_Ptr( fluid_sub, (real)0.0, x, y, z, Time[lv], 0.0, lv, NULL );
```

Comment:
It's good to set `Emag=0.0`.

But what I meant is that this function here always takes `AuxArray[] = NULL;`
Flu_ResetByUser_Func_Ptr( fluid_sub, (real)0.0, x, y, z, Time[lv], 0.0, lv, `NULL` );

So, I am worrying that,  in the future, if we use `Flu_ResetByUser_Func_Ptr` somewhere else with `AuxArray != NULL`.
`Flu_ResetByUser_Func_Ptr` will access `NULL` when it is calling from `Hydro_Init_ByFunction_AssignData`
and will give us a `segmentation fault`.

Reply:
@Inhyeok-Song `AuxArray[]` is user-defined. To use that, users must define their own `Flu_ResetByUser_API_Ptr` instead of using the default one. Note that `Flu_ResetByUser_API_Default()` also has `AuxArray = NULL`.

Reply:
@Inhyeok-Song My comment above is not fully correct. Even with a user-defined `Flu_ResetByUser_API_Ptr()`, `Hydro_Init_ByFunction_AssignData()` will still set `AuxArray=NULL` as you mentioned. So I think to use `AuxArray` one must also overwrite the default `Hydro_Init_ByFunction_AssignData()`, which is however not supported it. I'll file an issue about that. Thanks for the careful check!

Reply:
@Inhyeok-Song #153